### PR TITLE
feat: implement mongodb transactions

### PR DESCRIPTION
**Pull request by nordinh**
This new feature adds transaction support for MongoDB

Closes: #3051 

### Description of change

Adding support for transactions in MongoDB (transactions only work on replicasets).
By default, transactions are not enabled as is currently the default.
Transactions can be enabled through connection options.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [x] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md]

### Remark
Not added any new tests for this, as it requires quite a different MongoDB setup (replicaset/cluster).
Adding such a cluster in thee default docker-compose would make that a lot heavier.
Any suggestions on how to proceed?

### CODE REVIEW

**Comment by pleerock** [comment]
> Not added any new tests for this, as it requires quite a different MongoDB setup (replicaset/cluster).
Adding such a cluster in thee default docker-compose would make that a lot heavier.
Any suggestions on how to proceed?

Not sure about this one. We can skip tests for now.

Also, please merge latest master into your branch, since it contains fixes to test cases, to make sure all tests are running on your branch.

Also I need to review from people who use MongoDB before we can merge this PR.

Thanks for contribution.

**Comment by nordinh** [comment]
Hi @pleerock ,

Merged master, all tests seem to pass.

Kind regards,
Nordin

File: docs/mongodb.md [hunk 1]
```diff
@@ -404,3 +404,35 @@ Updates multiple documents within the collection based on the filter.
 #### `updateOne`
 
 Updates a single document within the collection based on the filter.
+
+
+## Using transactions
+
+Transactions are only supported on replicasets in MongoDB. Enabling transactions for mongodb can be done through connection parameters:
```
Comment (Ginden) [inline docs/mongodb.md:411]: `mongodb` - use `MongoDB`.

**Comment by pleerock** [comment]
I cannot really review mongodb features, because it's community-lead driver and I personally don't work a lot with mongodb. I need community help to review mongodb-related PRs. There are many people contributing to mongodb driver, but none of them willing to review other people's mongodb PRs.

**Comment by pleerock** [comment]
adding @mptr in the case of interest.

**Comment by pleerock** [comment]
Looks like no community interest here. Closing for now.

**Pull request closed without merging**